### PR TITLE
Added ColorOverlay and Shadow properties to BackStackParallax

### PR DESCRIPTION
**Pull request by aashay-gaikwad**
## Description

Added `ColorOverlay` and `Shadow` motion properties and added to `ParallaxBackstack` to better simulate the iOS behavior.
...

### CODE REVIEW

**Comment by KovalevAndrey** [comment]
@aashay-gaikwad could you include the video of how it looks now?

**Comment by aashay-gaikwad** [comment]
> @aashay-gaikwad could you include the video of how it looks now?
Yes!

https://github.com/bumble-tech/appyx/assets/74541941/b9620d65-24e9-4690-9f02-3726c30d5c05

File: appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/BackStackParallax.kt [hunk 1]
```diff
@@ -23,6 +26,7 @@ class BackstackParallax<InteractionTarget : Any>(
     defaultAnimationSpec = defaultAnimationSpec,
 ) {
     private val width = uiContext.transitionBounds.widthDp.value
+    private val slowInFastOutEasing = CubicBezierEasing(1f, 0f, 1f, 0f)
```
Comment (zsoltk) [inline appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/BackStackParallax.kt]: I guess this is rather `slowOutFastIn` (or even `delayedOutFastIn`)

Comment (manuel-martos) [inline appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/BackStackParallax.kt]: Isn't this easing to dramatic? https://cubic-bezier.com/#1,0,1,0
`FastOutSlowInEasing` doesn't work well here?

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/ColorOverlay.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    class Target(
+        val value: Float,
+        val easing: Easing? = null,
+    ) : MotionProperty.Target
+
+    override fun calculateRenderValue(base: Float, displacement: Float): Float =
+        base - displacement
+
+    override val visibilityMapper: ((Float) -> Boolean) = { true }
```
Comment (manuel-martos) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/ColorOverlay.kt:40]: Shouldn't visibility be calculated as it is for `Alpha` motion property?

File: appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/BackStackParallax.kt (hunk 1 above)
Comment (aashay-gaikwad) [reply appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/BackStackParallax.kt]: Oh right! Renamed

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/Shadow.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    class Target(
+        val value: Float,
+        val easing: Easing? = null,
+    ) : MotionProperty.Target
+
+    override fun calculateRenderValue(base: Float, displacement: Float): Float =
+        base - displacement
+
+    override val visibilityMapper: ((Float) -> Boolean) = { true }
```
Comment (manuel-martos) [inline appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/Shadow.kt:39]: Does shadow value equals to 0 means that there is no shadow present?

**Review by manuel-martos — APPROVED** [review]

File: appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/BackStackParallax.kt (hunk 1 above)
Comment (aashay-gaikwad) [reply appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/BackStackParallax.kt]: Yes, that is intentional, the shadow is to be shown until the element is scrolled to the end and then disappear immediately after.

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/ColorOverlay.kt (hunk 2 above)
Comment (aashay-gaikwad) [reply appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/ColorOverlay.kt:40]: The overlay shouldn't affect the behavior of the node. When the visibility is based on alpha, the element becomes invisible when the alpha of the overlay becomes 0.

File: appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/Shadow.kt (hunk 3 above)
Comment (aashay-gaikwad) [reply appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/interactions/core/ui/property/impl/Shadow.kt:39]: Yes, that is correct

**Review by zsoltk — APPROVED** [review]

**Pull request merged**